File: backend1/src/main/java/hwr/backend1/game/model/SlotConfig.java
package hwr.backend1.game.model;

import java.util.List;
import java.util.Map;

public record SlotConfig(
        int reels,
        int rows,
        int paylines,
        int minBet,
        int maxBet,
        int defaultBet,
        List<Payline> paylineDefinitions,
        Map<SlotSymbol, Map<Integer, Integer>> paytable
) {
}

File: slotz/types/slotConfig.ts
export interface PaylineDefinition {
    id: number;
    name: string;
    rows: number[];
}

export interface SymbolPayout {
    symbol: string;
    payouts: Record<number, number>;
}

export interface SlotConfig {
    reels: number;
    rows: number;
    paylines: number;
    minBet: number;
    maxBet: number;
    defaultBet: number;
    paylineDefinitions: PaylineDefinition[];
    symbols: SymbolPayout[];
}